packages/packagers/html/src/HTMLPackager.js

```diff
@@ -47,13 +49,73 @@ export default new Packager({
     }, []);
 
     let {html} = await posthtml([
-      insertBundleReferences.bind(this, bundles)
+      insertBundleReferences.bind(this, bundles),
+      replaceInlineAssetContent.bind(this, bundleGraph, getBundleResult)
     ]).process(code);
 
     return {contents: html};
   }
 });
 
+async function getAssetContent(
+  bundleGraph: BundleGraph,
+  getBundleResult,
+  assetId
+): Promise<?Blob> {
+  let inlineBundle: ?Bundle;
+  bundleGraph.traverseBundles((bundle, context, {stop}) => {
+    let mainAsset = bundle.getMainEntry();
+    if (mainAsset && mainAsset.meta && mainAsset.meta.inlineKey === assetId) {
+      inlineBundle = bundle;
+      stop();
+    }
+  });
+
+  if (inlineBundle) {
+    const bundleResult = await getBundleResult(inlineBundle, bundleGraph);
+
+    return bundleResult.contents;
+  }
+
+  return null;
+}
+
+async function replaceInlineAssetContent(
+  bundleGraph: BundleGraph,
+  getBundleResult,
+  tree
+) {
+  const inlineNodes = [];
+  tree.walk(node => {
+    if (node.attrs && node.attrs['data-parcel-key']) {
+      inlineNodes.push(node);
+    }
+    return node;
+  });
+
+  for (let node of inlineNodes) {
+    let newContent = await getAssetContent(
+      bundleGraph,
+      getBundleResult,
+      node.attrs['data-parcel-key']
+    );
+
+    if (newContent != null) {
+      node.content = newContent;
+
+      if (node.attrs['parcel-should-delete-type'] === 'true') {
```

Comment:
Could this be done in the transformer directly?

Reply:
I had considered that. There is an odd edge where if the compiling messes us it changed the type and not the code. 

My logic for here was that the transformer did the gathering and the packager is doing mutations to the inline assets, but this line felt odd.